x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -486,10 +486,12 @@ void searchVersionedDocumentForSubject(Subject subject, ActionListener<Versioned
         });
     }
 
+    private static final Pattern VALID_LITERAL_USERNAME = Pattern.compile("^[a-zA-Z0-9][a-zA-Z0-9_-]{0,255}$");
```

Comment:
Do we want to support `_` here? It would be preferable if `_` was only ever used as a separator and not part of the username.

Reply:
It might be my misunderstanding. I thought this was your suggestion. I can drop both `_` and `-`. I don't see it's necessary to have `-` here once we drop `_`.

Reply:
Discussed offline: dash (`-`) is still needed because cloud file-realm has it in usernames.